pandas/core/groupby/groupby.py

```diff
@@ -2433,21 +2447,40 @@ def _get_cythonized_result(
             name = obj.name
             values = obj._values
 
+            if numeric_only and not is_numeric_dtype(values):
+                continue
+
             if aggregate:
                 result_sz = ngroups
             else:
                 result_sz = len(values)
 
-            result = np.zeros(result_sz, dtype=cython_dtype)
-            func = partial(base_func, result, labels)
+            if needs_2d:
+                result = np.zeros((result_sz, 1), dtype=cython_dtype)
+            else:
+                result = np.zeros(result_sz, dtype=cython_dtype)
+            func = partial(base_func, result)
+
             inferences = None
 
+            if needs_counts:
+                counts = np.zeros(self.ngroups, dtype=np.int64)
+                func = partial(func, counts)
+
             if needs_values:
                 vals = values
                 if pre_processing:
                     vals, inferences = pre_processing(vals)
+                if needs_2d:
+                    vals = vals.reshape((-1, 1))
```

Comment:
Are these not 2d to begin with?

Reply:
Prior to this line, vals is a 1d array, e.g. [1, 2, 3]. After, it is 2d, e.g. [[1], [2], [3]].

Reply:
Hmm is that a problem with this function though or in the caller? This is only for `std` right? I think the keyword itself isn't obvious at first glance

Reply:
I don't follow what you're asking in the first question. Currently this is only used for `std`. Happy to workshop the name - what about "needs_2d_values"?

Reply:
What other functions do you see using this is maybe a good question to ask then. I don't think the concept of `needs_2d` is immediately apparent to other developers so want to see if there's something more generic we can use

Reply:
Other than mad (which also uses var), I don't know of any other function that would benefit from switching to this. I don't see a way to be more generic.

Reply:
can you use atleast_2d